easybuild/tools/toolchain/toolchain.py

```diff
@@ -238,6 +238,21 @@ def is_system_toolchain(self):
         """Return boolean to indicate whether this toolchain is a system(/dummy) toolchain."""
         return is_system_toolchain(self.name)
 
+    def set_minimal_build_env(self):
+        """Set up a minimal build environment, by setting $CC, $CXX, $FC environment variables."""
+
+        # this is only relevant when using a system toolchain,
+        # for proper toolchains these variables will get set via the call to set_variables()
+        env_vars = {
+            'CC': 'gcc',
+            'CXX': 'g++',
```

Comment:
I'm wondering if we should enhance this a bit and check whether these commands are available before setting the corresponding environment variable, and print a warning if they're not.

For `gcc` and `g++` that makes sense, and it could help with pointing out that `gcc` and `g++` are expected to be there when the `system` compiler is used.

This doesn't make sense for `gfortran` though, since I don't think we build anything with the `system` compiler which would require a Fortran compiler... So maybe we shouldn't be setting `$FC` & co here at all?

Thoughts on this @casparvl and @ocaisa, since this is relevant in the context of EESSI, see #3398?

Reply:
@boegel I made a PR to your branch adding checks on the presence of these commands: https://github.com/boegel/easybuild-framework/pull/51

It is important to note that we have easyconfigs using `SYSTEM` that load GCC and friends as dependencies (not using the `GCC` toolchain), for instance the `foss` easyconfigs. My PR will pick the first binaries found in `$PATH`, so it will prefer those from EB over those from the system if the current easyconfig loads such modules.

Reply:
> This doesn't make sense for gfortran though, since I don't think we build anything with the system compiler which would require a Fortran compiler... So maybe we shouldn't be setting $FC & co here at all?

Is there any reason we should nót set `FC`? Not setting it is based on the assumption that we'll never - ever - use it. It seems likely, but then again, you know what they say about assumptions... That's why I'm wondering if there could be adverse effects to setting it? If the same assumption holds (i.e. we don't use it anyway), I guess setting it is equally fine. If we dó happen to use it in some dark corner case, setting it would probably just pick up on the compiler we envisioned.

Reply:
@boegel commented to me in a private chat: since `gfortran` is not always available on all systems, setting it could generate more issues than it solves.

I agree with that, and thus with the approach of not setting `FC`.